### DISCUSSION TITLE
character training unfortunately no longer works?

### DISCUSSION THREAD

**Discussion by pip-bln** [TheLastBen/fast-stable-diffusion#512] [Q&A]
I started a small project in which I wanted to train 20 faces of members to create funny pictures for a private catalog.
So far it worked quite well for my first 7 attempts and with a few tries I was able to generate a suitable image for each face.
Unfortunately, this no longer works at all. The result is only pixel mud and the faces are not recognized at all. 
So far my source material was 40 images in 512x52 pixel format, with 100 training steps as recommended. Now it says that the images should be trained 200 times each, which obviously does not work. The specification with the text encoder has always seemed quite random to me so far and unfortunately not very accurate usable. Therefore I had set it so far simply on 50 wich worked quite well. Now it works neither at 50 nor at 100. 
Is there possibly a chance to get it working again so that faces can be recognized and trained?

**Comment by TheLastBen** [comment]
If you use the same settings and instance images as before, I don't see why you wouldn't get the same results

**Reply by pip-bln** [reply]
That's what my question is about: What has changed?
Why are 200 training steps now recommended instead of the previous 100? And why doesn't it work with both?
Could you possibly recommend me the best settings for 40 images in 512x512 format, where the goal is to be able to generate a face as well as possible? I would be really happy if I could finish the project I started.

**Reply by TheLastBen** [reply]
one face is easy, set the text encoder to 100% and do 4000 steps, if many faces, then the diffusion model is the limitation because it merges instances from the same class, men together and women together, so you won't get accurate results when training on subjects from the same class

**Reply by pip-bln** [reply]
I always have only one face per model. And as mentioned, that has always worked well so far (so far for 7 models). For 40 pictures, as you said, 4000 training steps were needed and everything turned out ok. That's why my assumption was that something seems to have changed. - Maybe someone else here has made a similar experience?? I am actually faced with a mystery ^^

**Reply by DivinoAG** [reply]
I would also appreciate some clarification on the new recomendations. Just about a week ago, the recommendation was to use 100 training steps per images (@TheLastBen actually mentioned on Reddit you could use about half of that and get good results), and the default value for the text encoder was 35%. I trained a checkpoint with 30 images, 3000 steps, and 40% on the text encoder and had great results. 

Today I trained a model using the same settings, and the subject I trained barely features in some of the generated images, and now the recommendations for steps is doubled, the default value for the text encoder is 100%, and there's no explanation of why these things changed. It would be great if you could explain what changed and what does that mean for the training process.

**Reply by TheLastBen** [reply]
The previous settings were meant for a style, but for a face, the text encoder needs to be set to a higher value, and I recommended more steps because instance images quality varies, so to be safe, I changed it to 200s per image, but you can do less and get great results

**Reply by DivinoAG** [reply]
What I don't understand is that before those changes, the lower text enconder value (40%) and 100 steps per image worked fantastic for faces. But now it doesn't work anymore. For example, this is an image created with a model I created on the 5th of this month using these settings I just mentioned. Features great resemblance to the subject, really good style transfer, it's all around a great model.
![masterpiece_____symmetry_____a_portrait_of_irine_m_AAGNnFMc_RealESRGAN_x4plus](https://user-images.githubusercontent.com/5044664/202196494-0964d111-2cdf-4477-af3a-97728fb79845.jpeg)

Last night I tried to make another model using almost the same 30 images (I replaced a couple), same number of steps per image, same text encoder value, and the person barely comes across the image when using the exact same prompt, the style is completely overwhelming the final image.
![00005-masterpiece!!!, symmetry!!!, a portrait of irnmyr, highly detailed, trending on artstation, art by Carne Griffiths_3663330320](https://user-images.githubusercontent.com/5044664/202196831-296e229e-0f43-4af0-a64d-b1111bb6eacc.png)

It's not just the change in the recommendation, but something clearly is not working as it was before in the notebook itself, or at least we would be able to reproduce the exact procedure we used... last week. It would be helpful to understand what changed, so we can adjust our settings accordingly.

**Reply by TheLastBen** [reply]
from the 5th to today, I haven't changed anything related to the training, you can check the commit history. It is just your settings, don't the use "contain_faces" option

**Reply by DivinoAG** [reply]
Why not? Isn't that the purpose of that option?

**Reply by TheLastBen** [reply]
that option prevents the trained subject from leaking to other subjects, but it doesn't improve the result and it requires way more steps because it has 200 more pictures to train on.

**Reply by pip-bln** [reply]
Honestly, it looks similar to DivinoAG for me - So even if nothing has actually changed, it has definitely changed in the result. I had as mentioned with my first 7 faces absolutely no problems and now I suddenly do not get further. My impression is that faces are no longer recognized as well as before, or hardly at all, but instead more the style of the entire image.
